examples/counter.rs

```diff
@@ -0,0 +1,179 @@
+use std::{
+    net::SocketAddr,
+    sync::{Arc, Mutex},
+};
+
+use abci::{async_trait, types::*, Consensus, Info, Mempool, Server};
+
+/// Simple counter
+#[derive(Debug, Default, Clone)]
+pub struct CounterState {
+    block_height: i64,
+    app_hash: Vec<u8>,
+    counter: u64,
+}
+
+#[derive(Debug)]
+pub struct ConsensusConnection {
+    committed_state: Arc<Mutex<CounterState>>,
+    current_state: Arc<Mutex<Option<CounterState>>>,
+}
+
+impl ConsensusConnection {
+    pub fn new(
+        committed_state: Arc<Mutex<CounterState>>,
+        current_state: Arc<Mutex<Option<CounterState>>>,
+    ) -> Self {
+        Self {
+            committed_state,
+            current_state,
+        }
+    }
+}
+
+#[async_trait]
+impl Consensus for ConsensusConnection {
+    async fn init_chain(&self, _init_chain_request: InitChainRequest) -> InitChainResponse {
+        Default::default()
+    }
+
+    async fn begin_block(&self, _begin_block_request: BeginBlockRequest) -> BeginBlockResponse {
+        let committed_state = self.committed_state.lock().unwrap().clone();
```

Comment:
I think it's already locked in `Server`, so no need to protect here in application code?

Reply:
Can you point to the place where it is getting locked in `Server`?

Reply:
https://github.com/tendermint/rust-abci/pull/112/files#diff-4ce93534efc34e923ce01e975eb7ed80R177